From jic23@cam.ac.uk  Mon Dec  7 16:23:32 2009
From: Jonathan Cameron <jic23@cam.ac.uk>
Date: Sun, 22 Nov 2009 16:03:25 +0000
Subject: Staging: IIO: Add tsl2560-2 support to tsl2563 driver.
To: Greg KH <gregkh@suse.de>
Cc: Amit Kucheria <amit.kucheria@verdurent.com>
Message-ID: <4B0960CD.7020302@cam.ac.uk>

Minimal changes to driver. Just adds the device to the id
table and adjusts the Kconfig elements appropriately.

Adding further similar chips from TAOS is complicated by their
different conversion functions (and hence left for now).

Signed-off-by: Jonathan Cameron <jic23@cam.ac.uk>
Acked-by: Amit Kucheria <amit.kucheria@verdurent.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/iio/light/Kconfig   |    6 +++---
 drivers/staging/iio/light/tsl2563.c |   11 ++++++-----
 2 files changed, 9 insertions(+), 8 deletions(-)

--- a/drivers/staging/iio/light/Kconfig
+++ b/drivers/staging/iio/light/Kconfig
@@ -4,11 +4,11 @@
 comment "Light sensors"
 
 config SENSORS_TSL2563
-	tristate "TAOS TSL2563 ambient light sensor"
+	tristate "TAOS TSL256[0-3] ambient light sensor"
 	depends on I2C
 	help
-	 If you say yes here you get support for the Taos TSL2563
-	 ambient light sensor (found in N900).
+	 If you say yes here you get support for the Taos TSL2560,
+	 TSL2561, TSL2562 and TSL2563 ambient light sensors.
 
 	 This driver can also be built as a module.  If so, the module
 	 will be called tsl2563.
--- a/drivers/staging/iio/light/tsl2563.c
+++ b/drivers/staging/iio/light/tsl2563.c
@@ -38,8 +38,6 @@
 #include "../iio.h"
 #include "tsl2563.h"
 
-#define DRIVER_NAME  "tsl2563"
-
 /* Use this many bits for fraction part. */
 #define ADC_FRAC_BITS		(14)
 
@@ -738,14 +736,17 @@ out:
 }
 
 static const struct i2c_device_id tsl2563_id[] = {
-	{ DRIVER_NAME, 0 },
-	{ },
+	{ "tsl2560", 0 },
+	{ "tsl2561", 1 },
+	{ "tsl2562", 2 },
+	{ "tsl2563", 3 },
+	{}
 };
 MODULE_DEVICE_TABLE(i2c, tsl2563_id);
 
 static struct i2c_driver tsl2563_i2c_driver = {
 	.driver = {
-		.name	 = DRIVER_NAME,
+		.name	 = "tsl2563",
 	},
 	.suspend	= tsl2563_suspend,
 	.resume		= tsl2563_resume,
